### ISSUE TITLE
How to compile on Windows.

### ISSUE THREAD

**Issue by hpstep** [antirez/smallchat#10]
How to support compiling on windows? 
I use llvm-mingw or winlibs.

**Comment by resslr** [comment]
Try to get [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) (Windows Subsystem For Linux)

**Comment by resslr** [comment]
@hpstep Lmk when you installed WSL and if that doesn't work (You need to install the dependencies btw)

**Comment by superswan** [comment]
As stated by @statulr  you will need WSL to compile and run the program. The code is not portable to Windows since it is using libraries and system calls specific to *NIX systems. Socket programming on windows requires the winsock library. Maybe someone can make a Windows fork.

**Comment by MZalfred** [comment]
If you're looking to use LLVM-MinGW or WinLibs to compile code on Windows, it's a bit different from using the standard MinGW or GCC compilers. You need to grab the latest release of LLVM-MinGW on GitHub or on the official LLVM-MinGW website

**Comment by fl54321** [comment]
虚拟机上可以编译吗

**Comment by resslr** [comment]
> 虚拟机上可以编译吗

Yes as long as the virtual machine runs a compatible OS (Recommended: Ubuntu)

**Comment by pifeifei** [comment]
> 虚拟机上可以编译吗

cygwin 可以编译

编译后，这个怎么用啊

**Comment by resslr** [comment]
> > 虚拟机上可以编译吗
> 
> cygwin 可以编译
> 
> 编译后，这个怎么用啊

抱歉，如果这个翻译很糟糕，我不会说普通话，但基本上，因为它有一个 MAKEFILE，你可以执行命令smallchat，我很确定，或者你必须执行 GCC 或任何其他编译器，然后运行输出文件。

**Comment by resslr** [comment]
> > 虚拟机上可以编译吗
> 
> cygwin 可以编译
> 
> 编译后，这个怎么用啊

但老实说，我建议在 WSL 中执行此操作（之前已链接）

**Comment by pifeifei** [comment]
> > > 虚拟机上可以编译吗
> > 
> > 
> > cygwin 可以编译
> > 编译后，这个怎么用啊
> 
> 抱歉，如果这个翻译很糟糕，我不会说普通话，但基本上，因为它有一个 MAKEFILE，你可以执行命令smallchat，我很确定，或者你必须执行 GCC 或任何其他编译器，然后运行输出文件。

![image](https://github.com/antirez/smallchat/assets/45787170/b7d0a52e-cd4e-41c5-8476-a3543d34ae17)

cygwin 中装了 apt-cyg install netcat， 没有 nc 命令, 用 msys2 中的 nc 命令链接可以使用了

还有就是 这个里面的代码量很少，应该没有兼容问题， 也就用这个编译试试，结果成了

一直没用 wsl 是因为服务器用的 windows 2008 r2 和centos, 就没有 用 wsl 了

**Comment by resslr** [comment]
> cygwin 中装了 apt-cyg install netcat， 没有 nc 命令, 用 msys2 中的 nc 命令链接可以使用了
> 
> 还有就是 这个里面的代码量很少，应该没有兼容问题， 也就用这个编译试试，结果成了
> 
> 一直没用 wsl 是因为服务器用的 windows 2008 r2 和centos, 就没有 用 wsl 了

好吧，很高兴看到它有效！

**Comment by resslr** [comment]
[Full Conversation Transcribed To English](https://imperialb.in/q0pzzygv)

**Comment by mrchipset** [comment]
I have made a win32 porting at this [repo](https://github.com/mrchipset/smallchat-win32). You may have a try.
Util now, there are still some non-blocking console input issue, if some one have any idea, please help.

**Comment by resslr** [comment]
@antirez You can archive this issue now
